### DISCUSSION TITLE
Build a declarative test framework on CUE?

### DISCUSSION THREAD

**Discussion by tianzhou** [cue-lang/cue#1616] [Ideas] [answered]
I am inspired by [dagger](https://docs.dagger.io/) and thinking if it's appropriate to use CUE as the declarative format to define test cases.

The idea is to define the domain model and a workflow involving different actions (e.g. API action) as a CUE file. Then feed that file into a test driver to run the test.

**Comment by mpvl** [comment, marked as answer]
See https://github.com/cue-lang/cue/issues/209
